File: presentation/10_Advanced/advanced.py
# Topic: Advanced Global Scope
# Manage Global State

state = {
    "status": "Active",
    "users": 0
}

def main():
    global state
    
    # Python relies on dictionary access or global keyword
    print(f"Current Status: {state['status']}")
    
    state['status'] = "Maintenance"
    
    print(f"New Status: {state['status']}")

if __name__ == "__main__":
    main()

File: presentation/10_Advanced/advanced.swift
// Topic: Advanced Global Scope
// Manage Global State (Singleton / Global Var)

import Foundation

struct AppState {
    static var status = "Active"
    static var users = 0
}

print("Current Status: \(AppState.status)")

AppState.status = "Maintenance"

print("New Status: \(AppState.status)")
